detox/src/utils/getConfigurationFile.js

```diff
@@ -0,0 +1,26 @@
+const path = require('path');
+const packageJson = 'package.json';
+const detoxRc = '.detoxrc';
+
+function requireWithRoot(suffix) {
+  return require(path.resolve(suffix))
+}
+
+function getConfigurationFile(configPath) {
```

Comment:
Regarding the testing question, I struggle to see what's the issue here :confused: especially if you rewrite it in an OOP way, something like this:

```js
class ConfigurationResolver {
  constructor(cwd = process.cwd()) {
    this.cwd = cwd;
  }

  getDetoxConfiguration(configPath) {
    if (configPath) {
      return this.loadConfiguration(configPath);
    }

    const { detox } = this.loadConfiguration('package.json');
    if (detox) {
      return detox;
    }

    return this.loadConfiguration('.detoxrc');
  }

  loadConfiguration(configPath) {
    return require(path.resolve(this.cwd, configPath));
  }
}

ConfigurationResolver.default = new ConfigurationResolver();

module.exports = ConfigurationResolver;
```

You can instantiate it with any `cwd` directory and test against it.

Reply:
For the tests, I was referring to using the current testing framework which initializes detox (using `package.json` config) and runs the tests against it.
You're speaking about a unit test?